tests/test/pointee.rs

```diff
@@ -0,0 +1,112 @@
+use super::*;
+
+#[test]
+fn last_field_metadata() {
```

Comment:
Can you add some negative tests? Wrong metadata, something that can't be projected, etc?